### PR TITLE
pods: only track failed pods for retry-on-Update

### PR DESCRIPTION
**Pull request by dcbw**
Suggested by Girish.

@danwinship @squeed @girishmg

### CODE REVIEW

File: go-controller/pkg/ovn/ovn.go [hunk 1]
```diff
@@ -332,33 +332,34 @@ func podScheduledAndWantsNetwork(pod *kapi.Pod) bool {
...
-			if err := oc.addLogicalPort(pod); err != nil {
-				logrus.Errorf(err.Error())
+			if podScheduledAndWantsNetwork(pod) {
+				if err := oc.addLogicalPort(pod); err != nil {
+					logrus.Errorf(err.Error())
+					retryPods.Insert(string(pod.UID))
+				}
 			} else {
-				handledPods.Insert(string(pod.UID))
+				// Handled unscheduled pods later on Update
```
Comment (girishmg) [inline go-controller/pkg/ovn/ovn.go]: NIT: s/handled/handle/

File: go-controller/pkg/ovn/ovn.go [hunk 2]
```diff
@@ -332,33 +332,34 @@ func podScheduledAndWantsNetwork(pod *kapi.Pod) bool {
...
+				retryPods.Insert(string(pod.UID))
```
Comment (girishmg) [inline go-controller/pkg/ovn/ovn.go]: @dcbw was about to comment on this, but your latest changes captures it :-).

Only caveat here is that the retryPods now will also have pods with HostNetwork, set?. Perhaps @L346 we need to wrap the Insert with `if !pod.spec.Hostnetwork`?

**Comment by dcbw** [comment]
@girish how about now?

**Pull request merged**